game/yodesk.ksy

```diff
@@ -0,0 +1,414 @@
+meta:
+  id: yodesk
+  application: "Star Wars: Yoda Stories"
+  file-extension: dta
+  license: CC0-1.0
+  ks-version: 0.9
+  endian: le
+  encoding: ASCII
+seq:
+  - id: catalog
+    type: catalog_entry
+    repeat: until
+    repeat-until: _.type == "ENDF"
+types:
+  catalog_entry:
+    seq:
+      - id: type
+        type: str
+        size: 4
+      - id: content
+        type:
+          switch-on: type
+          cases:
+             '"VERS"': version
+             '"STUP"': setup_image
+             '"CHAR"': characters
+             '"CAUX"': character_auxiliaries
+             '"CHWP"': character_weapons
+             '"PUZ2"': puzzles
+             '"SNDS"': sounds
+             '"TILE"': tiles
+             '"TNAM"': tile_names
+             '"ZONE"': zones
+             '"ENDF"': endf
+             _: unknown_case_error
+  unknown_case_error:
+    seq:
+      - id: message
+        contents: "Value did not match any expected type"
+  version:
+    seq:
+      - id: version
+        type: u4
+  setup_image:
+    seq:
+      - id: size
+        type: u4
+      - id: pixels
+        size: size
+  sounds:
+    seq:
+      - id: size
+        type: u4
+      - id: count
+        type: s2
+      - id: sounds
+        type: prefixed_strz
+        repeat: expr
+        repeat-expr: -count
+  tile_names:
+    seq:
+      - id: size
+        type: u4
+      - id: names
+        type: tile_name
+        repeat: until
+        repeat-until:  _.tile_id == -1
+  tile_name:
+    seq:
+      - id: tile_id
+        type: s2
+      - id: name
+        type: str
+        size: 0x18
+        if: tile_id != -1
+  tiles:
+    seq:
+      - id: size
+        type: u4
+      - id: tiles
+        type: tile
+        repeat: expr
+        repeat-expr: size / ( 32 * 32 + 4)
```

Comment:
```sizeof<tile>```, but not the best way. A better way is `size: size` + `repeat: eos`

Reply:
I'm now using `repeat: eos`, thanks for the tip. Adding an additional layer to get the sub-stream right took me a little while (even though it's properly document). Accessing a specific tile from code is a little ridiculous with all the nested structures:

<img width="279" alt="Screenshot 2021-01-24 at 17 22 31" src="https://user-images.githubusercontent.com/697021/105636706-d13dd700-5e69-11eb-9531-dec598773ba6.png">

```javascript
const tileCategory = catalog.entries.find(entry => entry.type === "TILE");

tileCategory.content.tiles.tiles.tiles[xx];
```